src/Workspaces/Core/Portable/Utilities/SerializableBytes.cs

```diff
@@ -85,7 +85,77 @@ internal static PooledStream CreateReadableStream(Stream stream, CancellationTok
                     int chunkOffset = 0;
                     while (bytesRemainingForThisChunk > 0)
                     {
-                        int bytesRead = await stream.ReadAsync(chunk, chunkOffset, bytesRemainingForThisChunk, cancellationToken).ConfigureAwait(false);
+                        // In many .NET Framework versions (specifically the 4.5.* series, but probably much earlier
+                        // and also later) there is this particularly interesting bit in FileStream.BeginReadAsync:
```

Comment:
Yeah, this is https://github.com/dotnet/corefx/issues/6007 for corefx.  I agree it should be fixed.
cc: @JeremyKuhne, @ianhays, @danmosemsft